### ISSUE TITLE
TinyBert : Why using several Linear layer / activations to get the logits ?

### ISSUE THREAD

**Issue by astariul** [huawei-noah/Pretrained-Language-Model#14]
In TinyBert, you get the prediction logits by applying **ReLu activation** followed by **Linear layer** :

https://github.com/huawei-noah/Pretrained-Language-Model/blob/05462fed9247cc0fc12e741f05e574caa25a3e87/TinyBERT/transformer/modeling.py#L1135

**But why is this necessary ?**

Because `pooled_output` is already a **Linear layer** followed by an **activation function** :

https://github.com/huawei-noah/Pretrained-Language-Model/blob/05462fed9247cc0fc12e741f05e574caa25a3e87/TinyBERT/transformer/modeling.py#L534-L537

**Comment by nlpBeginner** [comment]
> In TinyBert, you get the prediction logits by applying **ReLu activation** followed by **Linear layer** :
> 
> https://github.com/huawei-noah/Pretrained-Language-Model/blob/05462fed9247cc0fc12e741f05e574caa25a3e87/TinyBERT/transformer/modeling.py#L1135
> 
> **But why is this necessary ?**
> 
> Because `pooled_output` is already a **Linear layer** followed by an **activation function** :
> 
> https://github.com/huawei-noah/Pretrained-Language-Model/blob/05462fed9247cc0fc12e741f05e574caa25a3e87/TinyBERT/transformer/modeling.py#L534-L537

Thank you for the helpful review! The results reported in our paper are obtained by the activations of 'tanh + relu' and we will test the results with only the activation of 'tanh'.

**Comment by chuanhuayang** [comment]
But how did you come up with this  idea 'tanh + relu'？what's your motivation?
